### PR TITLE
[m4] add new port

### PR DESCRIPTION
**Pull request by AnticliMaxtic**
**Describe the pull request**
Add support for m4 https://www.gnu.org/software/m4/m4.html

- #### Which triplets are supported/not supported? 
  linux

- Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  yes

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  yes

### CODE REVIEW

File: ports/m4/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+set(VERSION 1.4.19)
+set(SHA512 f5dd0f02fcae65a176a16af9a8e1747c26e9440c6c224003ba458d3298b777a75ffb189aee9051fb0c4840b2a48278be4a51d959381af0b1d627570f478c58f2)
+# cmake-format: off
+vcpkg_download_distfile(ARCHIVE
+    URLS "https://ftp.gnu.org/gnu/m4/m4-${VERSION}.tar.gz"
+    FILENAME "m4-${VERSION}.tar.gz"
+    SHA512 ${SHA512}
+)
+
+if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux")
```
Comment (Neumann-A) [inline ports/m4/portfile.cmake:11]: a) VCPKG_TARGET_IS_LINUX
b) vcpkg is not supposed to provide tools
c) so it only makes sense to add this port if vcpkg plans to replace msys. So this  needs to work on windows.

File: scripts/ci.baseline.txt [hunk 2]
```diff
@@ -785,6 +785,14 @@ luasocket:x64-linux=fail
...
 luasocket:x64-uwp=fail
 lzfse:arm-uwp=fail
+m4:arm64-windows         = skip
+m4:arm-uwp               = skip
+m4:x64-osx               = skip
+m4:x64-uwp               = skip
+m4:x64-windows           = skip
+m4:x64-windows-static    = skip
+m4:x64-windows-static-md = skip
+m4:x86-windows           = skip
```
Comment (autoantwort) [inline scripts/ci.baseline.txt:795]: You don't have to skip triplets that are already excluded by the support statement :)

**Comment by LilyWangLL** [comment]
@AnticliMaxtic Can you address the review suggestions? Thanks!

**Comment by AnticliMaxtic** [comment]
@LilyWangLL I am happy to address these, but as per @Neumann-A  comment, it seems unlikely that this MR will be approved anyway? The reason I have submitted this request is in an attempt to get libsystemd packaged as it is a requirement for sdbus-cpp (https://github.com/microsoft/vcpkg/pull/20212) so in order to properly complete that request I believe that I need this one as well. I am open to suggestions, and am new enough to trying to contribute to this project that any guidance would be appreciated. 

I am coming from a strong background in the use of (forgive my use of a bad word) Conan and these tools/packages are already available there so was attempting to create a parity with that workflow

File: scripts/ci.baseline.txt (hunk 2 above)
Comment (AnticliMaxtic) [reply scripts/ci.baseline.txt]: removed this change

File: ports/m4/portfile.cmake (hunk 1 above)
Comment (AnticliMaxtic) [reply ports/m4/portfile.cmake]: removed call to `VCPKG_CMAKE_SYSTEM_NAME`

Comment (dg0yt) [reply ports/m4/portfile.cmake]: > this needs to work on windows.

:+1:
Why not even try?

File: ports/m4/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+vcpkg_configure_make(
+  SOURCE_PATH "${SOURCE_PATH}"
+  AUTOCONFIG
+  OPTIONS
+)
+# cmake-format: on
+
+vcpkg_install_make()
+
+vcpkg_copy_pdbs()
```
Comment (JackBoosY) [inline ports/m4/portfile.cmake:25]:
```suggestion
vcpkg_copy_pdbs()

vcpkg_copy_tools(AUTO_CLEAN)
```

File: ports/m4/vcpkg.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "m4",
+  "version": "1.4.19",
+  "description": "GNU m4 is an implementation of the traditional Unix macro processor.",
+  "homepage": "https://github.com/autotools-mirror/m4",
+  "supports": "linux"
```
Comment (JackBoosY) [inline ports/m4/vcpkg.json:6]: For Windows, we always use mingW and msys provides these tools. So I think we can accept this.

**Review by JackBoosY — APPROVED** [review]

**Review by JackBoosY — CHANGES_REQUESTED** [review]
One minor changes.

File: ports/m4/portfile.cmake (hunk 3 above)
Comment (LilyWangLL) [reply ports/m4/portfile.cmake:25]: This tools has been installed correct directory and this port only support `linux`, so not need `vcpkg_copy_tools` here.

**Review by LilyWangLL — APPROVED** [review]

**Review by JackBoosY — APPROVED** [review]

File: ports/m4/vcpkg.json (hunk 4 above)
Comment (dg0yt) [reply ports/m4/vcpkg.json:6]: IMO, this port should not be limited to Linux, or not be added at all.

> For Windows, we always use mingW and msys provides these tools. So I think we can accept this.

And for Linux, it is typically taken from the system.
And for macOS, it is typically taken from Brew.

**Comment by ras0219-msft** [comment]
>forgive my use of a bad word

Competing solutions are not bad words here -- we all share the goal of making C++ the best language for developers that it can be!

>... I believe that I need this one as well

Unlike some other package managers, vcpkg does distinguish between libraries you need for the target (and therefore _need_ to be built from source using your target toolchain) versus tools that you only use in service of building those libraries. This enables us to use prebuilt tooling either from the system package manager or from some other project (such as MSYS2). This is how we currently acquire m4 for all of the existing autoconf-based libraries in vcpkg.

Assuming your need for m4 is entirely because of autoconf, instead take a look at `vcpkg_configure_make()`[1] which will automatically find, acquire, and run an appropriate autotools toolchain for any platform (including Windows).

[1] https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_configure_make.md

**Comment by LilyWangLL** [comment]
> 

Ping @AnticliMaxtic for response.

**Comment by LilyWangLL** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

**Pull request closed without merging**